### ISSUE TITLE
classloader leak with playframework dev run

### ISSUE THREAD

**Issue by ornicar** [kamon-io/Kamon#629]
`kamon.Kamon.counter("foo")` causes a classloader leak in playframework when using the dev workflow with `run`.

Here's a minimal project with instructions to reproduce the issue: https://github.com/ornicar/play-cdi-leak/tree/kamon

My setup:
- Linux 5.4.2
- openjdk version "13.0.1" 2019-10-15
- scala 2.13.1
- playframework 2.8.0
- sbt 1.3.5

[edit] Same issue on playframework side: https://github.com/playframework/playframework/issues/9935

**Comment by ivantopo** [comment]
Hey @ornicar, we released a new version of the SBT plugin that clears up a part of Kanela that was retaining most of the memory so far. It is not a definitive solution, but it should improve the situation. Please give it a try, it is version [2.0.6](https://github.com/kamon-io/sbt-kanela-runner/releases/tag/v2.0.6) of the sbt-kanela-runner plugin.

**Comment by ornicar** [comment]
Hello, thanks for that, but I don't use the kanela runner.

**Comment by ivantopo** [comment]
Hey @ornicar! 

In case this is still bothering you, we just released Kamon `2.1.11` with a few changes that should fix the memory leak you were experiencing with Play in development mode. I tried it with the example you provided and it does fix the issue! Although it is necessary to apply a small change to the application loader to call `Kamon.stop()`:

```scala
class MyApplicationLoader extends ApplicationLoader {
  def load(context: ApplicationLoader.Context): Application = {
    // This block is new and necessary.
    context.lifecycle.addStopHook { () =>
      Kamon.stop()
    }

    new MyComponents(context).application
  }
}
```
Thanks to @jtjeferreira for making this possible!
